### DISCUSSION TITLE
Issue with vignette trial: crawlWrap

### DISCUSSION THREAD

**Discussion by meg-abytes** [bmcclintock/momentuHMM#121] [Q&A] [answered]
I have tried using the elephant example provided in the September 2, 2021 vignette:

```
URL <- paste0("https://www.datarepository.movebank.org/bitstream/handle/",
              "10255/move.373/Elliptical%20Time-Density%20Model%20%28Wall%",
              "20et%20al.%202014%29%20African%20Elephant%20Dataset%20%",
              "28Source-Save%20the%20Elephants%29.csv")
rawData <- read.csv(url(URL))
rawData <- rawData[,c(11,3,4,5,6)]
colnames(rawData) <- c("ID","time","lon","lat","temp")
# only keep first track
rawData <- subset(rawData,ID==unique(ID)[1])

# convert times from factors to POSIX
rawData$time <- as.POSIXct(rawData$time,tz="GMT")
# project to UTM coordinates using package rgdal
library(rgdal)
llcoord <- SpatialPoints(rawData[,3:4],
                         proj4string=CRS("+proj=longlat +datum=WGS84"))
utmcoord <- spTransform(llcoord,CRS("+proj=utm +zone=30 +datum=WGS84"))
# add UTM locations to data frame
rawData$x <- attr(utmcoord,"coords")[,1]
rawData$y <- attr(utmcoord,"coords")[,2]

# fit crawl model
ln.prior <- function(theta) dnorm(theta[2],-4,2,log=TRUE)
crwOut <- crawlWrap(rawData, 
                    timeStep="hour",
                    theta=c(6.855,-0.007),
                    fixPar=c(NA,NA))
```
 

At which point I get the following error message:
Error in crawlWrap(rawData, timeStep = "hour", theta = c(6.855, -0.007),  : 
  crawl::crwMLE failed for all individuals.  Check crawl::crwMLE arguments and/or consult crawl documentation.

I have also tried 
`crwOut <- crawlWrap(obsData=rawData, timeStep="hour", retryFits=10)`

With the same error message

As I am just using the data supplied in the vignette URL I am not sure what is going wrong.  I have the latest version of momentuHMM  (‘1.5.4’) and crawl (‘2.2.1’)

I am trying to use this package on my own data, but as an R neophyte, I am trying to run through the examples first to make sure
I understand how to apply it to my own data. 

Thank you!

**Comment by bmcclintock** [comment, marked as answer]
I am unable to reproduce the error, which suggest to me there may be issues with your installations of `momentuHMM` and/or `crawl`.  I'd suggest uninstalling `momentuHMM` and `crawl`, restarting your machine, and then reinstalling `momentuHMM`. If that doesn't fix the issue, you could try calling `crawl::crwMLE` directly:
```
crawl::crwMLE(data=rawData, theta=c(6.855, -0.007), fixPar=c(NA,NA))
```
or run `crawlWrap` in `debug` mode to see where/why the error is occurring:
```
debug(crawlWrap)
crwOut <- crawlWrap(rawData, 
                    timeStep="hour",
                    theta=c(6.855,-0.007),
                    fixPar=c(NA,NA))
```

**Reply by meg-abytes** [reply]
Thank you for your reply. 

I tried this several times over the past week and finally got it to work this morning! My guess is that I had some packages in R that needed to be updated in order for this one to work- although which ones I needed were not immediately obvious. I am on my way to trying it with my own data and excited to see what I find. Thanks for creating the package and the vignette. So far, both have been very helpful :)
